web/src/components/vis/mixins/axisPlot.js

```diff
@@ -11,46 +10,72 @@ function labelAxis(label, msg, xFunc, yFunc, rot) {
 }
 
 export const axisPlot = {
-  methods: {
-    axisPlot({
-      margin, xrange, yrange, xlabel, ylabel, duration,
-    }) {
-      // Collect necessary props.
-      const {
-        width,
-        height,
-      } = this.$props;
-
-      // Grab the root SVG element.
-      const svg = select(this.$refs.svg);
-      this.svg = svg;
-
-      // Compute the size of the data rectangle.
-      this.margin = margin;
-
-      const dwidth = width - margin.left - margin.right;
-      this.dwidth = dwidth;
-
-      const dheight = height - margin.top - margin.bottom;
-      this.dheight = dheight;
-
-      // Create X and Y scales.
-      const scaleX = scaleLinear()
+  props: {
+    width: {
+      type: Number,
+      required: true,
+    },
+    height: {
+      type: Number,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      axisPlotInitialized: false,
+      margin: {
+        top: 20,
+        right: 0,
+        bottom: 50,
+        left: 50,
+      },
+      xrange: [-1.2, 1.2],
+      yrange: [-1.2, 1.2],
+      xlabel: 'x axis',
+      ylabel: 'y axis',
+      duration: 500,
+    };
+  },
+  computed: {
+    dwidth() {
+      const { width, margin } = this;
+      return width - margin.left - margin.right;
+    },
+    dheight() {
+      const { height, margin } = this;
+      return height - margin.top - margin.bottom;
+    },
```

Comment:
These two computed properties are FANTASTIC. I am mad that I didn't think to do this (still getting my head around Vue, obviously :-D).